src/features/Faucet/stateFactory.tsx

```diff
@@ -0,0 +1,107 @@
+import { FaucetService } from '@services/ApiService/Faucet';
+import { StoreAccount } from '@types';
+import { TUseStateReducerFactory } from '@utils';
+
+import { FaucetState, ITxFaucetResult } from './types';
+
+const FaucetFactory: TUseStateReducerFactory<FaucetState> = ({ state, setState }) => {
```

Comment:
Sorry if it wasn't clear. But I would prefer we use a reducer over stateFactory.
As an example: https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/features/SendAssets/SendAssets.reducer.tsx

Reply:
Actually, after further thought it might be ok to use a stateFactory for this for now.